### PR TITLE
Add filters for api v2 filtering

### PR DESCRIPTION
**Pull request by bbonislawski**

### CODE REVIEW

**Review by VygovskySergey — COMMENTED** [review]
@bbonislawski  for me looks fine, maybe add some empty lines after guard clauses, what do you think?

**Comment by bbonislawski** [comment]
@VygovskySergey done!

File: core/lib/spree/core/filters/comparable.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        end
+
+        def gteq(base_scope, value)
+          return base_scope unless value
+
+          base_scope.where(arel_attribute.gteq(value))
+        end
+
+        def lt(base_scope, value)
+          return base_scope unless value
```
Comment (kushniryb) [inline core/lib/spree/core/filters/comparable.rb]: Empty line

File: core/lib/spree/core/filters/comparable.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          base_scope.where(arel_attribute.lt(value))
+        end
+
+        def lteq(base_scope, value)
+          return base_scope unless value
```
Comment (kushniryb) [inline core/lib/spree/core/filters/comparable.rb]: Empty line

File: core/lib/spree/core/filters/comparable.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+module Spree
+  module Core
+    module Filters
+      class Comparable
+        def initialize(arel_attribute:)
+          @arel_attribute = arel_attribute
+        end
+
+        def call(base_scope:, filter:)
+          base_scope = gt(base_scope, filter[:gt])
```
Comment (kushniryb) [inline core/lib/spree/core/filters/comparable.rb]: Can we perhaps call it just `scope`?

File: core/lib/spree/core/filters/text.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+module Spree
+  module Core
+    module Filters
+      class Text
+        def initialize(arel_attribute:)
+          @arel_attribute = arel_attribute
```
Comment (kushniryb) [inline core/lib/spree/core/filters/text.rb]: Can we call it simply `attribute`?

File: core/lib/spree/core/filters/date.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+module Spree
+  module Core
+    module Filters
+      class Date < Comparable
```
Comment (kushniryb) [inline core/lib/spree/core/filters/date.rb]: https://github.com/rubocop-hq/ruby-style-guide#single-line-classes
What do you think?
I'm not a big fan of `Class.new(something)` but we can write it as `class Date < Comparable; end` to make it more compact

**Review by kushniryb — APPROVED** [review]
LGTM 👍

File: core/lib/spree/core/filters/date.rb (hunk 5 above)
Comment (bbonislawski) [reply core/lib/spree/core/filters/date.rb]: I think it's good in situation where you have more code in module. In this case, it doesn't look too good in my opinion

```
module Spree
  module Core
    module Filters
      class Number < Comparable; end
    end
  end
end
```

We don't need compact code in this case imo

File: core/lib/spree/core.rb [hunk 6]
```diff
@@ -81,6 +81,7 @@ class DestroyWithOrdersError < StandardError; end
 require 'spree/service_module'
 
 require 'spree/core/importer'
+require 'spree/core/filters'
```
Comment (damianlegawiec) [inline core/lib/spree/core.rb]: @bbonislawski `Filters` are kind of ambiguous - maybe sth like `CollectionFilters` ?

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: core/lib/spree/core.rb (hunk 6 above)
Comment (bbonislawski) [reply core/lib/spree/core.rb]: Changed

**Comment by bbonislawski** [comment]
@damianlegawiec approve?

**Review by damianlegawiec — APPROVED** [review]
👍

**Pull request merged**